### PR TITLE
Event listeners leak warning (tiny-jsonrpc)

### LINKED ISSUE #104: Event Listener Leak
In an web app I am making I continually spawn and destroy spooky instances using the built in ".destory()" method.

This works, but every eleventh spooky respawn I get a double EventEmitter memleak warning (see below for raw dump).

Poking around the spooky and tiny-json-rpc sources pinned this line in tiny-json-rpc as the source of the error:
```
        this._server.stream.on('data', this._onData.bind(this)); //Line 36 Of stream-client
        this._server.stream.on('drain', this._onDrain.bind(this));  //Line 37
```

What I think is happening is that when the spooky object is destroyed it doesn't clear these listeners from its _rpcClient object, thus causing these leaks.

I could be way off base, but I'm pretty sure that's what's going on.

Error:

```
(node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at RequestStream.EventEmitter.addListener (events.js:160:15)
    at new StreamClient (/Users/jldrogo/Developer/repo/Crawler/node_modules/spooky/node_modules/tiny-jsonrpc/lib/tiny-jsonrpc/stream-client.js:36:29)
    at new Spooky (/Users/jldrogo/Developer/repo/Crawler/node_modules/spooky/lib/spooky.js:75:27)
    at resetSpooky (/Users/jldrogo/Developer/repo/Crawler/worker.js:45:14)
    at Promise.<anonymous> (/Users/jldrogo/Developer/repo/Crawler/links.js:83:20)
    at Promise.<anonymous> (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/node_modules/mpromise/lib/promise.js:177:8)
    at Promise.EventEmitter.emit (events.js:95:17)
    at Promise.emit (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/node_modules/mpromise/lib/promise.js:84:38)
    at Promise.fulfill (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/node_modules/mpromise/lib/promise.js:97:20)
    at Promise.resolve (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/lib/promise.js:108:15)
(node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at RequestStream.EventEmitter.addListener (events.js:160:15)
    at new StreamClient (/Users/jldrogo/Developer/repo/Crawler/node_modules/spooky/node_modules/tiny-jsonrpc/lib/tiny-jsonrpc/stream-client.js:37:29)
    at new Spooky (/Users/jldrogo/Developer/repo/Crawler/node_modules/spooky/lib/spooky.js:75:27)
    at resetSpooky (/Users/jldrogo/Developer/repo/Crawler/worker.js:45:14)
    at Promise.<anonymous> (/Users/jldrogo/Developer/repo/Crawler/links.js:83:20)
    at Promise.<anonymous> (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/node_modules/mpromise/lib/promise.js:177:8)
    at Promise.EventEmitter.emit (events.js:95:17)
    at Promise.emit (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/node_modules/mpromise/lib/promise.js:84:38)
    at Promise.fulfill (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/node_modules/mpromise/lib/promise.js:97:20)
    at Promise.resolve (/Users/jldrogo/Developer/repo/Crawler/node_modules/mongoose/lib/promise.js:108:15)
```

### PR DESCRIPTION
**Pull request by noodny**
Fixes #104

### CODE REVIEW

File: lib/spooky.js [hunk 1]
```diff
@@ -102,14 +102,17 @@ function Spooky(options, callback) {
     serializeMethods(options.casper);
 
     if (options.child.transport === 'http') {
-        this._child =
-          Spooky._instances[options.child.port] = this._spawnChild();
+        this._child = Spooky._instances[options.port] = this._spawnChild();
```
Comment (lawnsea) [inline lib/spooky.js]: Why did this change from `options.child.port` to `options.port`?

Comment (noodny) [inline lib/spooky.js]: Must've been a copy/paste issue. That should naturally be `options.child.port`.

**Comment by lawnsea** [comment]
Thanks, @noodny. I will test this later this week and land it.

**Comment by lukaszblacha** [comment]
@lawnsea could you merge this PR?

**Pull request closed without merging**